polkadot/node/core/approval-voting/src/criteria.rs

```diff
@@ -170,14 +169,38 @@ fn relay_vrf_modulo_cores(
 		);
 	}
 
-	vrf_in_out
-		.make_bytes::<BigArray>(approval_types::v2::CORE_RANDOMNESS_CONTEXT)
-		.0
-		.chunks_exact(4)
-		.take(num_samples as usize)
-		.map(move |sample| CoreIndex(u32::from_le_bytes(clone_into_array(&sample)) % max_cores))
-		.unique()
-		.collect::<Vec<CoreIndex>>()
+	if 2 * num_samples > max_cores {
+		gum::error!(
+			target: LOG_TARGET,
+			n_cores = max_cores,
+			num_samples,
+			max_modulo_samples = MAX_MODULO_SAMPLES,
+			"Suboptimal configuration `num_samples` should be less than `n_cores` / 2",
+		);
+	}
```

Comment:
How small to small testnets go?  We need to make sure this works for them?

Reply:
If you're worried about small testnets, then this works too:
```
use rand::seq::SliceRandom;
let mut random_cores = (0..max_cores).collect::<Vec<Coreindex>>();
random_cores.partial_shuffle(rand_chacha,num_samples);
random_cores.rev();
random_cores.truncate(num_samples);
```

ref. https://docs.rs/rand/latest/src/rand/seq/mod.rs.html#594-613

Reply:
I've a better idea:
```
let num_samples = max(1, min( num_samples, max_cores/2 ));
```
I think this covers small testnets, but makes the sampling work for real deployments.

Reply:
Done, added also some tests, I think we are safe now we should be able to catch any changes in the underlaying implementation if any might happen.